lib/simulator/simulatormodule.c

```diff
@@ -952,7 +952,25 @@ static PyObject *deregister_callback(PyObject *self, PyObject *args)
     gstate = TAKE_GIL();
 
     pSihHdl = PyTuple_GetItem(args, 0);
-    hdl = (gpi_sim_hdl)PyLong_AsUnsignedLong(pSihHdl);
```

Comment:
Regarding python and long, Python PyLong objects are not guaranteed to be 4/8 bytes for 32/64 bit machins. They are arbitrary size integers. From a python script you can put a value larger than 2^64 in a PyLong and the PyLong_AsUnsigned will return -1 and raise an error when called.